tools/audio/aifc.c

```diff
@@ -0,0 +1,552 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (C) 2024 ZeldaRET
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "aifc.h"
+#include "util.h"
+
+#define CC4_CHECK(buf, str) \
+    ((buf)[0] == (str)[0] && (buf)[1] == (str)[1] && (buf)[2] == (str)[2] && (buf)[3] == (str)[3])
+
+#define CC4(c1, c2, c3, c4) (((c1) << 24) | ((c2) << 16) | ((c3) << 8) | (c4))
+
+#define FREAD(file, data, size)                                                                      \
+    do {                                                                                             \
+        if (fread((data), (size), 1, (file)) != 1) {                                                 \
+            error("[%s:%d] Could not read %lu bytes from file", __FILE__, __LINE__, (size_t)(size)); \
+        }                                                                                            \
+    } while (0)
```

Comment:
I don't want things outside of sampleconv to depend on sampleconv, and vice-versa

Reply:
After talking offline, it seems like the reasoning here is:
* Ease of releasing sampleconv separately as a standalone tool, although that is not currently a goal
* Historical reasons, sampleconv was written to replace z64sample which itself was standalone

Personally I think it would be fine to integrate it better with the rest of the audio tools, since releasing sampleconv separately is speculative at this point, and duplicating some 600-700 lines of code seems pretty sad. But I won't push it